apollo-router/src/plugins/connectors/form_encoding.rs

```diff
@@ -0,0 +1,155 @@
+use itertools::Itertools;
+use serde_json_bytes::ByteString;
+use serde_json_bytes::Map;
+use serde_json_bytes::Value;
+
+pub(super) fn encode_json_as_form(value: &Map<ByteString, Value>) -> String {
+    let mut encoded: form_urlencoded::Serializer<String> =
+        form_urlencoded::Serializer::new(String::new());
+
+    fn print_json_value_literal(value: &Value) -> String {
```

Comment:
I feel like there's a way to avoid all the `to_string` and `format!` by embedding this in `recurse`. I'm gonna play with it and send it to you if it works.